### PR TITLE
F/fix filename parse

### PR DESCRIPTION
**Pull request by AwaNoodle**
Found some issues with parsing **filename:** (wasn't supported and causes a segfault) and **commit:** (was relying on a 16 char hash which doesn't seem to always be the case). 

I've tested this on my CVS repos I'm converting and they're passing thru now.

### CODE REVIEW

File: cvsclone.l [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- * 
+ *
```
Comment (akavel) [inline cvsclone.l:19]: The copious trailing whitespace changes make the PR unreadable for me when viewed in the github's UI, unfortunately. Could you please move them to a separate PR, or at least a separate commit?

**Comment by akavel** [comment]
By the way, if you're planning to do more with your fork, and/or "take stewardship" of the project, I'm happy to put any "redirect info" in my repo pointing to your one as "main". Just saying in case you're interested.

File: cvsclone.l (hunk 1 above)
Comment (AwaNoodle) [inline cvsclone.l:19]: Aye, you're right. I'll bin the current revision and resubmit without the line ending changes
